sponsorship_compassion/migrations/10.0.1.0.7/pre-migration.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Nicolas Badoux <n.badoux@hotmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    if not version:
+        return
+
+    for partner in env['res.partner'].search(
+            ['partner.title.gender', '!=', False]):
+        if partner.title.name not in ['Madam', 'Mister', 'Misters', 'Ladies']:
```

Comment:
This condition could be added in the search criterias to avoid making a big loop and testing each time. Then you can as well simply call `upsert_constituent` on the recordset, avoiding to have to make a loop.